galley/pkg/config/source/kube/apiserver/status/controller.go

```diff
@@ -157,10 +170,26 @@ mainloop:
 			continue mainloop
 		}
 
+		// Get the map of status objects. If it doesn't already exist, create it.
+		statusObj, ok := u.Object["status"]
+		if !ok {
+			statusObj = make(map[string]interface{})
```

Comment:
It looks like you can avoid this allocation in certain cases (e.g. desiredStatus == nil). Which may save from allocating a large number of small objects.

Reply:
We'd only allocate if the status field was entirely blank already, which should normally only trigger when this runs for the first time for a resource. So I don't think it will be expensive.

Reply:
It means you'd be allocating a map for each resource that has a new message on it. It can add up in a large cluster. It looks like it would be cheap to avoid the allocation by slightly restructuring the code here.